src/LEGO.AsyncAPI.Bindings/Sns/Principal.cs

```diff
@@ -0,0 +1,66 @@
+namespace LEGO.AsyncAPI.Bindings.Sns;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using LEGO.AsyncAPI.Models;
+using LEGO.AsyncAPI.Models.Interfaces;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+
+public class Principal : IAsyncApiElement
+{
+    public Principal(AsyncApiAny value)
+    {
+        this.Value = value;
+    }
+
+    public AsyncApiAny Value { get; }
```

Comment:
to be honest, I would have liked that it was limited to specific classes.
AsyncApiAny, can be.. well anything - so from an API standpoint, the user has no idea what to put there.
We could do a marker interface IPrincipalValue or whatever and make the classes needed

Reply:
You can make an interface and 3 structures that can satisfy it.
One for the list, one for the object and one for the value. Serialize/Deserialize as needed.

That should be perfectly fine. 

We have a StringOrList type already, so that should help as well.

Reply:
Ok I've created an interface and two structs to satisfy it. Let me know if you've got any questions or comments 👍